addons/l10n_in_withholding/models/account_move.py

```diff
@@ -196,31 +211,32 @@ def _is_section_applicable(section_alert, threshold_sums, invoice_currency_rate,
-                warning = set()
-                commercial_partner_id = move.commercial_partner_id
-                existing_section = (self.l10n_in_withhold_move_ids.line_ids + move.line_ids).tax_ids.l10n_in_section_id
-                for section_alert, lines in _group_by_section_alert(move.invoice_line_ids).items():
-                    if (
-                        (section_alert not in existing_section
-                        or [line for line in lines if section_alert not in line.tax_ids.l10n_in_section_id])
-                        and move._l10n_in_is_warning_applicable(section_alert)
-                        and _is_section_applicable(
+        if self.country_code == 'IN' and self.move_type in ['in_invoice', 'out_invoice']:
+            warning = set()
+            commercial_partner_id = self.commercial_partner_id
+            existing_section = (self.l10n_in_withhold_move_ids.line_ids + self.line_ids).tax_ids.l10n_in_section_id
+            for section_alert, lines in _group_by_section_alert(self.invoice_line_ids).items():
+                if (
+                    (section_alert not in existing_section
+                    or self._get_tcs_applicable_lines(lines))
+                    and self._l10n_in_is_warning_applicable(section_alert)
+                    and _is_section_applicable(
+                        section_alert,
+                        self._get_sections_aggregate_sum_by_pan(
                             section_alert,
-                            move._get_sections_aggregate_sum_by_pan(
-                                section_alert,
-                                commercial_partner_id
-                            ),
-                            move.invoice_currency_rate,
-                            lines
-                        )
-                    ):
-                        warning.add(section_alert.id)
-                warning_sections = self.env['l10n_in.section.alert'].browse(warning)
-                if warning_sections:
-                    move.l10n_in_tcs_tds_warning = warning_sections._get_warning_message()
-                else:
-                    move.l10n_in_tcs_tds_warning = False
-            else:
-                move.l10n_in_tcs_tds_warning = False
+                            commercial_partner_id
+                        ),
+                        self.invoice_currency_rate,
+                        lines
+                    )
+                ):
+                    warning.add(section_alert.id)
+            warning_sections = self.env['l10n_in.section.alert'].browse(warning)
+            return warning_sections
+
+    def _get_tcs_applicable_lines(self, lines):
+        tcs_applicable_lines = self.env['account.move.line']
+        for line in lines:
+            if line.section_id not in line.tax_ids.l10n_in_section_id:
+                tcs_applicable_lines |= line
+        return tcs_applicable_lines
```

Comment:
```suggestion
        tcs_applicable_lines = set()
        for line in lines:
            if line.section_id not in line.tax_ids.l10n_in_section_id:
                tcs_applicable_lines.add(line.id)
        return self.env['account.move.line'].browse(tcs_applicable_lines)
```
Try to avoid operation on records as much as possible